src/Parsers/FunctionSecretArgumentsFinder.h

```diff
@@ -538,6 +542,42 @@ class FunctionSecretArgumentsFinder
             markSecretArgument(2);
     }
 
+    void findAzureBlobStorageTableEngineSecretArguments()
+    {
+       /// azureBlobStorageCluster('cluster_name', 'conn_string/storage_account_url', ...) has 'conn_string/storage_account_url' as its second argument.
+        size_t url_arg_idx = is_cluster_function ? 1 : 0;
+
+        if (isNamedCollectionName(0))
+        {
+            /// azureBlobStorage(named_collection, ..., account_key = 'account_key', ...)
+            if (maskAzureConnectionString(-1, true, 1))
+                return;
+            findSecretNamedArgument("account_key", 1);
+            return;
+        }
+
+        if (maskAzureConnectionString(url_arg_idx))
+            return;
+
+        /// We should check other arguments first because we don't need to do any replacement in case of
+        /// azureBlobStorage(connection_string|storage_account_url, container_name, blobpath, format, [account_name, account_key, ...])
+        /// azureBlobStorageCluster(cluster, connection_string|storage_account_url, container_name, blobpath, format, [account_name, account_key, ...])
+        size_t count = function->arguments->size();
+        if ((url_arg_idx + 4 <= count) && (count <= url_arg_idx + 7))
+        {
+            String fourth_arg;
+            if (tryGetStringFromArgument(url_arg_idx + 3, &fourth_arg))
+            {
+                if (fourth_arg == "auto" || KnownFormatNames::instance().exists(fourth_arg))
+                    return; /// The argument after 'url' is a format: s3('url', 'format', ...)
```

Comment:
this comment is wrong (I see it is copied from https://github.com/ClickHouse/ClickHouse/blob/d4adcaf0b7565f8b6baa0eab603353903123f24d/src/Parsers/FunctionSecretArgumentsFinder.h#L246 but could you please anyway fix it)
I'd remove it at all, because this case is explained by comment above in lines 562-564.